pkg/clustermesh/kvstoremesh/kvstoremesh.go

```diff
@@ -134,17 +143,56 @@ func RegisterSyncWaiter(p SyncWaiterParams) {
 	)
 }
 
+func (km *KVStoreMesh) run() {
+	defer km.wg.Done()
+	km.wg.Add(1)
+
+	for {
+		km.logger.Info("Locking backend lock")
+		lock, err := km.backend.LockPath(km.ctx, backendLockName)
+		if err != nil {
+			errTimer := time.NewTimer(1 * time.Second)
+			select {
+			case <-km.ctx.Done():
+				return
+			case <-errTimer.C:
+				continue
+			}
+		}
+		km.logger.Info("Locked backend lock")
+
+		km.common.Start(km.ctx)
+
+		select {
+		case <-km.ctx.Done():
+			km.common.Stop(km.ctx)
+			lock.Unlock(km.ctx)
+			return
+		// TBD: are we the only reader of the channel?
+		case e := <-km.backend.StatusCheckErrors():
+			km.logger.With(e).Info("Synchronization stopped")
+			km.common.Stop(km.ctx)
```

Comment:
A common approach is to trigger a full restart of the container when leader election is lost. I'd probably do the same here, just to make sure to restart from a clean state (you can use `hive.Shutdowner` for that).